src/main/java/org/isf/sms/service/SmsSender.java

```diff
@@ -57,77 +56,50 @@ public void run() {
 			}
 			if (!smsList.isEmpty()) {
-				LOGGER.info("Found {} SMS to send", smsList.size());
-				if (SmsParameters.MODE.equals("GSM")) {
-					
-					SmsSenderGSM sender = new SmsSenderGSM();
-					if (sender.initialize()) {
-						for (Sms sms : smsList) {
-							if (sms.getSmsDateSched().before(new Date())) {
-								boolean result = sender.sendSMS(sms, GeneralData.DEBUG);
-								if (result) {
-									sms.setSmsDateSent(new Date());
-									try {
-										smsOp.saveOrUpdate(sms);
-									} catch (OHServiceException ohServiceException) {
-										LOGGER.error(ohServiceException.getMessage(), ohServiceException);
-									}
-									LOGGER.debug("Sent");
-								} else {
-									LOGGER.error("Not sent");
+				LOGGER.info("Found {} SMS to send", Integer.valueOf(smsList.size()));
+				SmsSenderOperations sender = Context.getApplicationContext().getBean(SmsSenderOperations.class);
+				if (sender.initialize()) {
+					for (Sms sms : smsList) {
+						if (sms.getSmsDateSched().before(new Date())) {
+							boolean result = sender.sendSMS(sms);
+							if (result) {
+								sms.setSmsDateSent(new Date());
+								try {
+									smsOp.saveOrUpdate(sms);
+								} catch (OHServiceException e) {
+									e.printStackTrace();
 								}
+								LOGGER.debug("Sent");
+							} else {
+								LOGGER.error("Not sent");
 							}
 						}
-						sender.terminate();
-					} else {
-						LOGGER.error("SMS Sender GSM initialization error");
-						LOGGER.error("Stopping SMS Sender...");
-						setRunning(false);
-					}
-					
-				} else if (SmsParameters.MODE.equals("HTTP")) {
-					
-					SmsSenderHTTP sender = new SmsSenderHTTP();
-					if (sender.initialize()) {
-						for (Sms sms : smsList) {
-							if (sms.getSmsDateSched().before(new Date())) {
-								boolean result = sender.sendSMS(sms, GeneralData.DEBUG);
-								if (result) {
-									sms.setSmsDateSent(new Date());
-									try {
-										smsOp.saveOrUpdate(sms);
-									} catch (OHServiceException ohServiceException) {
-										LOGGER.error(ohServiceException.getMessage(), ohServiceException);
-									}
-									LOGGER.debug("Sent");
-								} else {
-									LOGGER.error("Not sent");
-								}
-							}
-						}
-					} else {
-						LOGGER.error("SMS Sender HTTP initialization error");
-						LOGGER.error("Stopping HTTP Sender...");
-						setRunning(false);
 					}
+					boolean terminationResult = sender.terminate();
+					LOGGER.debug("termination result: " + terminationResult);
+				} else {
+					LOGGER.error("SMS Sender HTTP initialization error");
+					LOGGER.error("Stopping HTTP Sender...");
+					setRunning(false);
 				}
 			} else {
 				LOGGER.debug("No SMS to send.");
 			}
 			try {
 				Thread.sleep(delay * 1000);
-			} catch (InterruptedException interruptedException) {
-				LOGGER.error(interruptedException.getMessage(), interruptedException);
+			} catch (InterruptedException e) {
+				e.printStackTrace();
```

Comment:
the correct form is with `interruptedException` and `LOGGER.error`

Reply:
done